benchmark/utils/preconditioners.hpp

```diff
@@ -0,0 +1,212 @@
+
+#ifndef GKO_BENCHMARK_UTILS_PRECONDITIONERS_HPP_
+#define GKO_BENCHMARK_UTILS_PRECONDITIONERS_HPP_
+
+
+#include <ginkgo/ginkgo.hpp>
+
+
+#include <map>
+#include <string>
+
+
+#include <gflags/gflags.h>
+
+
+#include "benchmark/utils/general.hpp"
+#include "benchmark/utils/overhead_linop.hpp"
+
+
+DEFINE_string(
+    preconditioners, "none",
+    "A comma-separated list of preconditioners to use. "
+    "Supported values are: none, jacobi, parict, parilu, parilut, ilu, "
+    "parict-isai, parilu-isai, parilut-isai, ilu-isai, overhead");
+
+DEFINE_uint32(parilu_iterations, 5,
+              "The number of iterations for ParICT/ParILU(T)");
```

Comment:
```suggestion
              "The number of iterations for ParIC(T)/ParILU(T)");
```

Reply:
This one is deliberate, since we don't have a ParIC implementation right now